transactions/src/main/java/org/ehcache/transactions/xa/internal/xml/TxCacheManagerServiceConfigurationParser.java

```diff
@@ -31,36 +30,32 @@
 import javax.xml.transform.Source;
 import javax.xml.transform.stream.StreamSource;
 
-import java.io.IOException;
 import java.net.URI;
-import java.net.URL;
+import java.util.Map;
+import java.util.function.Supplier;
 
+import static java.util.Collections.singletonMap;
 import static org.ehcache.core.util.ClassLoading.delegationChain;
 import static org.ehcache.transactions.xa.internal.TypeUtil.uncheckedCast;
 
 /**
  * @author Ludovic Orban
  */
 @Component
-public class TxCacheManagerServiceConfigurationParser extends BaseConfigParser<LookupTransactionManagerProviderConfiguration> implements CacheManagerServiceConfigurationParser<TransactionManagerProvider> {
-  private static final URI NAMESPACE = URI.create("http://www.ehcache.org/v3/tx");
-  private static final URL XML_SCHEMA = TxCacheManagerServiceConfigurationParser.class.getResource("/ehcache-tx-ext.xsd");
+public class TxCacheManagerServiceConfigurationParser extends BaseConfigParser<LookupTransactionManagerProviderConfiguration>
+  implements CacheManagerServiceConfigurationParser<TransactionManagerProvider, LookupTransactionManagerProviderConfiguration> {
+  private static final String NAMESPACE = "http://www.ehcache.org/v3/tx";
   public static final String TRANSACTION_NAMESPACE_PREFIX = "tx:";
   private static final String TRANSACTION_ELEMENT_NAME = "jta-tm";
   private static final String TRANSACTION_LOOKUP_CLASS = "transaction-manager-lookup-class";
 
   @Override
-  public Source getXmlSchema() throws IOException {
-    return new StreamSource(XML_SCHEMA.openStream());
+  public Map<URI, Supplier<Source>> getSchema() {
+    return singletonMap(URI.create(NAMESPACE), () -> new StreamSource(getClass().getResourceAsStream("/ehcache-tx-ext.xsd")));
```

Comment:
What happens if the schema isn't located?

Reply:
Ultimately it will blow up with an NPE when trying to assemble the complete schema object. That would indicate a packaging failure on our side though.

Reply:
It might be best to throw at this point ... waiting until schema assembly, the underlying cause of the error (missing a specific schema file) is likely to be unclear in the NPE that is ultimately issued.